docs/public/using/backend/rpcapi-reference.md

```diff
@@ -1436,6 +1439,151 @@ If the `batch_id` is undefined the following error is returned:
 }
 ```
 
+
+### API method: `batch_status`
+
+This method replaces deprecated method
+[API method: get_batch_job_result](#api-method-get_batch_job_result).
+
+Returns the number of waiting, running and finished *tests*. Optionally it also
+returns the [*test ids*][Test id] of the *batch test*.
+
+Example valid request and response, respectively, where no [*test ids*][Test id]
+were requested:
+
+```json
+{
+    "jsonrpc": "2.0",
+    "id": 147559211994909,
+    "method": "batch_status",
+    "params": {
+        "batch_id": "8"
+    }
+}
+```
+
+```json
+{
+   "jsonrpc": "2.0",
+   "id": 147559211994909,
+   "result": {
+      "finished": 5,
+      "running": 5,
+      "waiting": 195
+   }
+}
+```
+
+Example valid request and response, respectively, where [*test ids*][Test id]
+were requested for all three status values:
+
+```json
+{
+    "jsonrpc": "2.0",
+    "id": 147559211994909,
+    "method": "batch_status",
+    "params": {
+        "batch_id": "8",
+        "list_running": true,
+        "list_finished": true,
+        "list_waiting": true
+    }
+}
+```
+
+Finished [*test id*][Test id] objects were requested (no finished objects
+existing):
+
+```json
+{
+   "jsonrpc": "2.0",
+   "id": 147559211994909,
+   "result": {
+      "finished": 0,
+      "finished_test_ids": [
+      ],
+      "runninng": 5,
+      "waiting": 195
+   }
+}
+```
+
+Finished [*test id*][Test id] objects were requested:
+
+```json
+{
+   "jsonrpc": "2.0",
+   "id": 147559211994909,
+   "result": {
+      "finished": 1,
+      "finished_test_ids": [
+         "43b408794155324b",
+      ],
+      "waiting_test_ids": [
+         "be9cbb44fff0b2a8",
+         "62f487731116fd87",
+      ],
+      "running_test_ids": [
+         "692f8ffc32d647ca",
+         "6441a83fcee8d28d"
+      ],
+      "waiting": 2,
+      "running": 2
+   }
+}
+```
+
+#### `"params"`
+
+An object with the property:
+
+* `"batch_id"`: A [*batch id*][Batch id], required.
```

Comment:
What is the type of the "batch_id" parameter ? String or Integer ?

Reply:
It's specified as a number in the "Batch id" data type specification (higher up in the document), but in the examples they're shown as strings. This discrepancy should probably be fixed.

Reply:
I have just followed how "batch_id" is shown in other examples.

Reply:
After some digging around, it turns out that Zonemaster::Backend::Validator allows batch_ids that are either plain non-negative numbers or strings containing non-negative numbers. However, it will correctly reject:
 - negative numbers or strings that parse to a negative number;
 - booleans, nulls and objects.

Be aware that JSON numbers map to double-precision floats: some integers cannot be appropriately represented as a JSON number type. It’s therefore safer, and I think perfectly appropriate, to use number-strings in this interface.

My suggestion: document `batch_id` better and state clearly that both numbers and number-strings are allowed.